clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java

```diff
@@ -1595,6 +1596,33 @@ public void testDescribeConfigsUnrequested() throws Exception {
         }
     }
 
+    @Test
+    public void testDescribeClientMetricConfigs() throws Exception {
+        ConfigResource metricResource0 = new ConfigResource(ConfigResource.Type.CLIENT_METRICS, "metric0");
+        ConfigResource metricResource1 = new ConfigResource(ConfigResource.Type.CLIENT_METRICS, "metric1");
+        try (AdminClientUnitTestEnv env = mockClientEnv()) {
+            env.kafkaClient().setNodeApiVersions(NodeApiVersions.create());
+            env.kafkaClient().prepareResponse(new DescribeConfigsResponse(
+                    new DescribeConfigsResponseData().setResults(asList(
+                            new DescribeConfigsResponseData.DescribeConfigsResult()
+                                    .setResourceName(metricResource0.name())
+                                    .setResourceType(metricResource0.type().id())
+                                    .setErrorCode(Errors.NONE.code())
+                                    .setConfigs(emptyList()),
+                            new DescribeConfigsResponseData.DescribeConfigsResult()
+                                    .setResourceName(metricResource1.name())
+                                    .setResourceType(metricResource1.type().id())
+                                    .setErrorCode(Errors.NONE.code())
+                                    .setConfigs(emptyList())))));
+            Map<ConfigResource, KafkaFuture<Config>> result = env.adminClient().describeConfigs(asList(
+                    metricResource0,
+                    metricResource1)).values();
+            assertEquals(new HashSet<>(asList(metricResource0, metricResource1)), result.keySet());
+            result.get(metricResource0).get();
```

Comment:
Is this line and the following something that we're testing that we don't throw exceptions?

Reply:
it is not expecting an exception - so if it gets an exception, the test will show